.github/workflows/reusable_test.yml

```diff
@@ -0,0 +1,65 @@
+name: Reusable Test Workflow
+
+on:
+  workflow_dispatch:
+    inputs:
+      pinned_botorch:
+        required: true
+        type: boolean
+      minimal_dependencies:
+        required: false
+        type: boolean
+        default: false
+  workflow_call:
+    inputs:
+      pinned_botorch:
+        required: true
+        type: boolean
+      minimal_dependencies:
+        required: false
+        type: boolean
+        default: false
+
+jobs:
+  tests-and-coverage:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10"]
```

Comment:
Should we start testing against 3.11 (here and in other places)? Botorch does...

Reply:
We should but this is currently blocked on https://github.com/pytorch/torchx/issues/744